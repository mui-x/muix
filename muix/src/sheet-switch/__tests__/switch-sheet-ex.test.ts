import {
  atomizeSheet,
  atomizeRuleset,
  toClassNamesWithQuery
} from "reactxx-sheeter";
import { initPlatform, Shape, ts } from "reactxx-tests";

import { sheetSwitch_registerVariantHandler } from "../index";

sheetSwitch_registerVariantHandler();

export const createSheet = () =>
  (ts.sheet = {
    root: (ts.view = [
      {
        $web: [
          {
            ":hover": {
              color: "red",
              $switch: {
                isClosed: {
                  cursor: "pointer"
                }
              }
            }
          }
        ]
      }
    ]),
    label: {},
    webOnly: {},
    nativeOnly: {}
  });

/*
describe("SWITCH define sheet", () => {
  it("NATIVE", () => {
    initPlatform(false);
    const sheet = atomizeSheet<Shape>(createSheet());
`
##########################################
##########################################
#
#  NATIVE
#
##########################################
##########################################
`    
    expect(sheet).toMatchInlineSnapshot();
  });
  it("WEB", () => {
    initPlatform(true);
    const sheet = atomizeSheet<Shape>(createSheet());
`
##########################################
##########################################
#
#  WEB
#
##########################################
##########################################
`    
    expect(sheet).toMatchInlineSnapshot();
  });
});
*/

describe("SWITCH define sheet", () => {
  it("NATIVE", () => {
    initPlatform(false);
    const sheet = atomizeSheet<Shape>(createSheet());
    `
##########################################
##########################################
#
#  NATIVE
#
##########################################
##########################################
`;
    expect(sheet).toMatchInlineSnapshot(`Object {}`);
  });
  it("WEB", () => {
    initPlatform(true);
    const sheet = atomizeSheet<Shape>(createSheet());
    `
##########################################
##########################################
#
#  WEB
#
##########################################
##########################################
`;
    expect(sheet).toMatchInlineSnapshot(`
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".a:hover { color:red /*root[0]/$web[0]*/ }",
        ],
      },
      Object {
        "atomicArray": Array [
          ".b:hover { cursor:pointer /*root[0]/$web[0]/:hover/$switch.isClosed*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`);
  });
});
