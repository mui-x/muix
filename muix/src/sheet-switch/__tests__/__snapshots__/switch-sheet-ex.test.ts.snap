// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SWITCH define sheet NATIVE 1`] = `
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "padding: 22 /*.*/",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          "padding: 11 /*./$switch.isOpened*/",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          "padding: 33 /*.*/",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`;

exports[`SWITCH define sheet WEB 1`] = `
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".a { padding:22px /*.*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          ".b { padding:11px /*./$switch.isOpened*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          ".c { padding:33px /*.*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`;

exports[`SWITCH define sheet WEB with pseudo: WRONG RESULT 1`] = `
Object {
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".a { padding:55px /*.*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          ".b { padding:44px /*./$switch.isOpened*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
          Object {
            "case": "isOpened",
            "type": "$switch",
          },
        ],
      },
      Object {
        "atomicArray": Array [
          ".c { padding:66px /*.*/ }",
        ],
        "conditions": Array [
          Object {
            "case": "isClosed",
            "type": "$switch",
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`;
